### PR TITLE
Deprecate neighbor list tag public member type

### PR DESCRIPTION
**Pull request by dalg24**
Per our discussion, style is not conforming and we believe it is unnecessary

### CODE REVIEW

File: core/src/Cabana_Macros.hpp [hunk 1]
```diff
@@ -17,6 +17,10 @@
 #define CABANA_FUNCTION KOKKOS_FUNCTION
 #define CABANA_INLINE_FUNCTION KOKKOS_INLINE_FUNCTION
 #define CABANA_FORCEINLINE_FUNCTION KOKKOS_FORCEINLINE_FUNCTION
+#if __has_cpp_attribute( deprecated )
```
Comment (sslattery) [inline core/src/Cabana_Macros.hpp]: nice - C++14 right?

Comment (dalg24) [reply core/src/Cabana_Macros.hpp]: yes

**Comment by dalg24** [comment]
@streeve good catch.  Fixed.

**Comment by dalg24** [comment]
CI [passed](https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Cabana/detail/PR-207/5/pipeline/)
Not sure why it did not report back

@streeve Are you okay with this version?

**Comment by dalg24** [comment]
Retest this please

**Pull request merged**